src/library/scala/util/Either.scala

```diff
@@ -265,7 +265,7 @@ sealed abstract class Either[+A, +B] extends Product with Serializable {
    *  @param f The side-effecting function to execute.
    */
   def foreach[U](f: B => U): Unit = this match {
-    case Right(b) => f(b)
+    case Right(b) => f(b); ()
```

Comment:
I didn't consider the moral equivalence of "match with one case and empty default" and "unibranch if".